### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by robertbastian**
The description of test and CI jobs is out of date.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,25 +60,40 @@ need to run in order to recreate them.  These files may be run in more comprehen
...
- - `cargo make ci-job-test`: Runs `cargo test` on all the crates. This takes a while but is the main way of ensuring that nothing has been broken.
- - `cargo make ci-job-clippy`: Runs `cargo clippy` on all the crates.
- - `cargo doc --no-deps --all-features`: Recreates API docs locally; any warning should be fixed since it will be treated as an error in CI.
- - `cargo make ci-job-ffi`: Runs all of the FFI tests; mostly important if you're changing the FFI interface. This has several additional dependencies:
-     + [`Diplomat`](https://github.com/rust-diplomat/diplomat) installed at the appropriate version: `cargo make diplomat-install`
-     + `clang-15` and `lld-15` with the `gold` plugin (APT packages `llvm-15` and `lld-15`)
- - `cargo make ci-job-wasm`: Runs Rust-to-WASM tests. This also has a couple additional dependencies:
-     + Node.js version 16.18.0. This may not the one offered by the package manager; get it from the NodeJS website or `nvm`.
-     + [`Twiggy`](https://github.com/rustwasm/twiggy) (at least 0.7.0: `cargo install twiggy`
-     + [`emsdk`](https://github.com/emscripten-core/emsdk): Make sure to `./emsdk activate latest` it into your shell
```
Comment (sffc) [inline CONTRIBUTING.md:76]: Do we not use twiggy or emsdk (for `wasm-opt`) any more? I think we still need them.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,25 +60,40 @@ need to run in order to recreate them.  These files may be run in more comprehen
...
- - `cargo make ci-job-features`: This is a pretty slow test that tries to build all ICU4X crates with all feature combinations. It has an additional dependency:
-     + `cargo-all-features`: `cargo install cargo-all-features`
- - `cargo make ci-all`: This runs all of the tests above, and takes quite a while. It is usually preferred to run specific tests you think are relevant and let GitHub Actions run the full suite to uncover additional failures.
```
Comment (sffc) [inline CONTRIBUTING.md:79]: Do we still have `cargo make ci-all` or was it deleted?

File: CONTRIBUTING.md (hunk 1 above)
Comment (robertbastian) [reply CONTRIBUTING.md:76]: That's the binsize benchmark, we don't have cargo make commands for that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (robertbastian) [reply CONTRIBUTING.md:79]: It's still there

File: CONTRIBUTING.md (hunk 1 above)
Comment (robertbastian) [reply CONTRIBUTING.md:76]: *we don't have ci commands that are listed here

**Pull request merged**